src/cript/api/exceptions.py

```diff
@@ -129,6 +130,35 @@ def __str__(self) -> str:
         return error_message
 
 
+class CRIPTDuplicateNameError(CRIPTAPISaveError):
+    def __init__(self, api_response, json_data: str, parent_cript_save_error: CRIPTAPISaveError):
+        super().__init__(
+            parent_cript_save_error.api_host_domain, api_response["code"], api_response=api_response["error"], patch_request=parent_cript_save_error.patch_request, pre_saved_nodes=parent_cript_save_error.pre_saved_nodes, json_data=json_data
+        )
+
+        # We don't care if the data is invalid JSON
+        # So let's catch a couple of common exceptions and ensure still meaning error messages
+        # (and debug info in case it does happen.)
+        try:
+            json_dict = json.loads(self.json_data)
+        except (TypeError, json.JSONDecodeError):
+            self.name = "unknown_name"
+            self.node = "UnknownType"
+        try:
+            self.name = json_dict["name"]
+        except KeyError:
+            self.name = "unknown_name_key"
+        try:
+            self.node = json_dict["node"][0]
+        except KeyError:
+            self.node = "UnknownTypeKey"
+        except IndexError:
+            self.node = "UnknownTypeIdx"
+
+    def __str__(self) -> str:
+        return f"The name '{self.name}' for your {self.node} node is already present in CRIPT. Either choose a new name!"  # , or consider namespaces like 'MyNameSpace::{self.name}' instead."
```

Comment:
Good message!

Tells the user exactly what node had issues and how they can easily solve it!

Reply:
thx :)